src/node/fetch/fetch.node.test.ts

```diff
@@ -1,19 +1,31 @@
 import FormData from 'form-data';
 import { getLocal } from 'mockttp';
-import fetch from 'node-fetch';
+import type { Response } from 'node-fetch';
+import fetch, { Headers } from 'node-fetch';
 
 import { NetworkFetchError, RequestFetchError } from '../../core';
 import { MockTimers } from '../../mock';
 import { NodeTimers } from '../timers';
 import { NodeFetch } from './fetch.node';
 
-jest.mock('node-fetch');
+// Maybe this can be done better?
+// The issue here is that we need to mock the default export (fetch) while
+// preserving everything as as actual implementation.
+jest.mock('node-fetch', ()  => {
+  const actual = jest.requireActual('node-fetch');
+  
+  const base: any = jest.fn();
+  for (const [key, value] of Object.entries(actual)) {
+    base[key] = value;
+  }
+  base['default'] = base;
+
+  return base as unknown;
+});
```

Comment:
Not sure if it is better, but I usually do it this way:

```suggestion
jest.mock('node-fetch', () => {
  // eslint-disable-next-line @typescript-eslint/consistent-type-imports
  const actual = jest.requireActual<typeof import('node-fetch')>('node-fetch');

  return {
    __esModule: true,
    ...actual,
    default: jest.fn(),
  };
});
```

Reply:
As long as that works 😅 I wasn't sure at all how to mock this properly so if you usually do it this way let's stick with that.